sdk/eventhub/azure-eventhub/azure/eventhub/_eventprocessor/event_processor.py

```diff
@@ -185,33 +210,18 @@ def _create_tasks_for_claimed_ownership(self, claimed_partitions, checkpoints=No
                             event_received_callback,
                         ),
                     )
-                    if self._partition_initialize_handler:
-                        try:
-                            self._partition_initialize_handler(self._partition_contexts[partition_id])
-                        except Exception as err:  # pylint:disable=broad-except
-                            _LOGGER.warning(
-                                "EventProcessor instance %r of eventhub %r partition %r consumer group %r. "
-                                "An error occurred while running on_partition_initialize. The exception is %r.",
-                                self._id,
-                                self._partition_contexts[partition_id].eventhub_name,
-                                self._partition_contexts[partition_id].partition_id,
-                                self._partition_contexts[partition_id].consumer_group,
-                                err,
-                            )
-                            self._process_error(self._partition_contexts[partition_id], err)
-                    _LOGGER.info(
-                        "EventProcessor %r has claimed partition %r",
-                        self._id,
-                        partition_id
-                    )
+                    self._initialize_partition_consumer(partition_id)
 
     def _on_event_received(self, partition_context, event):
-        # type: (PartitionContext, EventData) -> None
-        with self._context(event):
-            if self._track_last_enqueued_event_properties:
-                partition_context._last_received_event = (  # pylint: disable=protected-access
-                    event
-                )
+        # type: (PartitionContext, Union[Optional[EventData], List[EventData]]) -> None
+        if event:
+            partition_context._last_received_event = (  # pylint:disable=protected-access
+                event[-1] if isinstance(event, List) else event
+            )
```

Comment:
I think it's more efficient to do this as a try/except:
```
try:
    partition_context._last_received_event = event[-1]
except TypeError:
    partition_context._last_received_event = event

Reply:
Changed to this. And it's indeed faster than if/esle in Python.